python/paddle/fluid/data_feeder.py

```diff
@@ -91,6 +90,22 @@ def check_type(input, input_name, expected_type, op_name, extra_message=''):
     # each step in dynamic graph mode, it will bring a heavy performance burden.
     if in_dygraph_mode():
         return
+
+    from .dygraph.dygraph_to_static.program_translator import in_declarative_mode
+    # NOTE: `in_declarative_mode` is used to determined whether this op is called under
+    # @declarative in transformation from dygrah to static layer. We add VarBase in
+    # expected_type to skip checking because varBase may be created and used in unusual way.
+    # Need a better design to be fix this.
+    if in_declarative_mode():
+        if not isinstance(expected_type, tuple):
+            expected_type = (expected_type, )
+        expected_type += (core.VarBase, )
+    elif isinstance(input, core.VarBase):
+        raise TypeError(
+            "Please use fluid.dygraph.guard() as context to run it in imperative Mode."
```

Comment:
or using `fluid.enable_dygraph` to switch to imperative mode firstly?

Reply:
Thank，done~